### PR TITLE
KAFKA-14253 - More informative logging

### PR DESCRIPTION
**Pull request by philipnee**
Includes 2 requirements from the ticket:
- Include the number of members in the group (I.e., "15 members participating" and "to 15 clients as")
- sort the member ids (to help compare the membership and assignment across rebalances)

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by philipnee** [comment]
Ideally, we want "1 member" or "n member**s**". But we don't need this much grammatical optimization.

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsPartitionAssignor.java [hunk 1]
```diff
@@ -637,8 +638,12 @@ private boolean assignTasksToClients(final Cluster fullMetadata,
                                                                    statefulTasks,
                                                                    assignmentConfigs);
 
-        log.info("Assigned tasks {} including stateful {} to clients as: \n{}.",
-                allTasks, statefulTasks, clientStates.entrySet().stream()
+        log.info("{} assigned tasks {} including stateful {} to clients as: \n{}.",
```
Comment (guozhangwang) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsPartitionAssignor.java]: nit: Assigned {} tasks including ... to {} clients as ..

**Review by guozhangwang — COMMENTED** [review]
Thanks @philipnee . Just one nit, otherwise LGTM.

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsPartitionAssignor.java (hunk 1 above)
Comment (philipnee) [reply streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsPartitionAssignor.java]: Ah, acked.

**Comment by philipnee** [comment]
Failures seem unrelated
```
Build / JDK 11 and Scala 2.13 / testReplicationIsCreatingTopicsUsingProvidedForwardingAdmin() – org.apache.kafka.connect.mirror.integration.MirrorConnectorsWithCustomForwardingAdminIntegrationTest
2m 15s
Build / JDK 11 and Scala 2.13 / testReplicationIsCreatingTopicsUsingProvidedForwardingAdmin() – org.apache.kafka.connect.mirror.integration.MirrorConnectorsWithCustomForwardingAdminIntegrationTest
2m 22s
Build / JDK 8 and Scala 2.12 / testCreatePartitionsUseProvidedForwardingAdmin() – org.apache.kafka.connect.mirror.integration.MirrorConnectorsWithCustomForwardingAdminIntegrationTest
2m 20s
Build / JDK 8 and Scala 2.12 / testConnectorReconfiguration – org.apache.kafka.connect.integration.ExactlyOnceSourceIntegrationTest
33s
Build / JDK 8 and Scala 2.12 / [1] Type=ZK, Name=testRegisterZkBrokerInKraft, MetadataVersion=3.4-IV0, Security=PLAINTEXT – kafka.server.KafkaServerKRaftRegistrationTest
8s
Build / JDK 8 and Scala 2.12 / [1] Type=ZK, Name=testRestartOldIbpZkBrokerInMigrationMode, MetadataVersion=3.3-IV0, Security=PLAINTEXT – kafka.server.KafkaServerKRaftRegistrationTest
4s
Build / JDK 8 and Scala 2.12 / [1] false – org.apache.kafka.streams.integration.SmokeTestDriverIntegrationTest
7m 15s
```

**Pull request merged**